src/aliceVision/image/dcp.hpp

```diff
@@ -0,0 +1,197 @@
+     */
+    typedef struct DCPProfileInfo {
+        bool has_color_matrix_1;
+        bool has_color_matrix_2;
+        bool has_forward_matrix_1;
+        bool has_forward_matrix_2;
+        bool has_look_table;
+        bool has_hue_sat_map;
+        bool has_tone_curve;
+        bool has_baseline_exposure_offset;
+
+        short light_source_1;
+        short light_source_2;
+        double temperature_1;
+        double temperature_2;
+
+        DCPProfileInfo()
+        {
+            has_color_matrix_1 = false;
+            has_color_matrix_2 = false;
+            has_forward_matrix_1 = false;
+            has_forward_matrix_2 = false;
+            has_look_table = false;
+            has_hue_sat_map = false;
+            has_tone_curve = false;
+            has_baseline_exposure_offset = false;
+
+            light_source_1 = 0;
+            light_source_2 = 0;
+            temperature_1 = 0.0;
+            temperature_2 = 0.0;
+        }
+
+    } DCPProfileInfo;
+
+    /**
+     * @brief A class representing a DCP color profile
+     */
+    class DCPProfile final
+    {
+    public:
+        DCPProfileInfo info;
+
+        using Triple = std::array<double, 3>;
+        using Matrix = std::array<Triple, 3>;
+
+        explicit DCPProfile(const std::string filename);
+        ~DCPProfile();
+
+        void apply(OIIO::ImageBuf& image, const DCPProfileApplyParams& params);
+        void apply(float* rgb, const DCPProfileApplyParams& params) const;
+
+    private:
+        struct HsbModify {
+            float hue_shift;
+            float sat_scale;
+            float val_scale;
+        };
+
+        struct HsdTableInfo {
+            int hue_divisions;
+            int sat_divisions;
+            int val_divisions;
+            int hue_step;
+            int val_step;
+            unsigned int array_count;
+            bool srgb_gamma;
+            struct {
+                float h_scale;
+                float s_scale;
+                float v_scale;
+                int max_hue_index0;
+                int max_sat_index0;
+                int max_val_index0;
+                int hue_step;
+                int val_step;
+            } pc;
+        };
+
+        void hsdApply(const HsdTableInfo& table_info, const std::vector<HsbModify>& table_base, float& h, float& s, float& v) const;
+
+        Matrix ws_sRGB;
+        Matrix sRGB_ws;
+        Matrix color_matrix_1;
+        Matrix color_matrix_2;
+        bool will_interpolate;
+        bool valid;
+        Matrix forward_matrix_1;
+        Matrix forward_matrix_2;
+        double baseline_exposure_offset;
+        std::vector<HsbModify> deltas_1;
+        std::vector<HsbModify> deltas_2;
+        std::vector<HsbModify> look_table;
+        HsdTableInfo delta_info;
+        HsdTableInfo look_info;
+
+        SplineToneCurve AS_tone_curve;
+
+        SplineToneCurve gammatab_srgb;
+        SplineToneCurve igammatab_srgb;
+    };
+
+}
```

Comment:
Add a comment `// namespace image` so that it's clear what scope this bracket is closing.